tst/com/amazon/corretto/crypto/provider/test/RsaCipherTest.java

```diff
@@ -518,13 +588,54 @@ public void native2jceOaepParams(final Integer keySize) throws GeneralSecurityEx
         nativeC.init(Cipher.ENCRYPT_MODE, keyPair.getPublic(), params);
         jceC.init(Cipher.DECRYPT_MODE, keyPair.getPrivate(), params);
 
-        final byte[] plaintext = getPlaintext(keySize / 8 - 42);
+        final int paddingSize = getPaddingSize(params.getParameterSpec(OAEPParameterSpec.class));
+        final byte[] plaintext = getPlaintext(keySize / 8 - paddingSize);
 
         final byte[] ciphertext = nativeC.doFinal(plaintext);
         final byte[] decrypted = jceC.doFinal(ciphertext);
         assertArrayEquals(plaintext, decrypted);
     }
 
+    @ParameterizedTest
+    @MethodSource("digestsXdigestsXlengthParams")
+    public void native2jceOaepParamSpecs(final String md1, final String md2, final Integer keySizeBits) throws GeneralSecurityException {
+        final int keySize = keySizeBits / 8;
+        final OAEPParameterSpec paramSpec = new OAEPParameterSpec(
+            md1,
+            "MGF1",
+            new MGF1ParameterSpec(md2),
+            PSource.PSpecified.DEFAULT
+        );
+        final int paddingSize = getPaddingSize(paramSpec);
+        assumeTrue(
+            keySize - paddingSize > 0,
+            String.format("Padding len %d greater >= key size %d", paddingSize, keySize)
+        );
+        final KeyPair keyPair = getKeyPair(keySizeBits);
+        final Cipher nativeC = getNativeCipher(OAEP_PADDING);
+        nativeC.init(Cipher.ENCRYPT_MODE, keyPair.getPublic(), paramSpec);
+
+        // verify that the OAEPParameterSpec is set properly.
+        final OAEPParameterSpec nativeSpec = nativeC.getParameters().getParameterSpec(OAEPParameterSpec.class);
+        assertOAEPParamSpecsEqual(paramSpec, nativeSpec);
+
+        final Cipher jceC = Cipher.getInstance(OAEP_PADDING);
+        jceC.init(Cipher.DECRYPT_MODE, keyPair.getPrivate(), paramSpec);
+        final byte[] plaintext = getPlaintext(keySize - paddingSize);
+        final byte[] ciphertext = nativeC.doFinal(plaintext);
+        final byte[] decrypted = jceC.doFinal(ciphertext);
+        assertArrayEquals(plaintext, decrypted);
```

Comment:
Can we please also test the inverse of this?

Reply:
yep, i'll rip this out and work on parameterizing native2Jce and jce2Native across the possible OAEPParameterSpec values.